src/server/identityutil/identityutil.go

```diff
@@ -0,0 +1,37 @@
+package identityutil
+
+import (
+	"bytes"
+
+	"github.com/pachyderm/pachyderm/v2/src/identity"
+
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/gogo/protobuf/types"
+
+	proto "github.com/gogo/protobuf/proto"
+)
+
+// PickConfig determines whether to use the config or jsonConfig.
+// JsonConfig should only be used for backwards compatibility.
+func PickConfig(config *types.Struct, jsonConfig string) ([]byte, error) {
+	switch {
+	case config != nil && config.Fields != nil:
+		marshaler := jsonpb.Marshaler{}
+		buf := bytes.Buffer{}
+		err := marshaler.Marshal(&buf, config)
+		if err != nil {
+			return nil, err
+		}
+		return buf.Bytes(), nil
+	case jsonConfig != "" && jsonConfig != "null":
+		return []byte(jsonConfig), nil
+	default:
+		return []byte(""), nil
```

Comment:
I think we should throw an error in this case

Reply:
Ahh you mentioned this before, but I missed, that you're right!